hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateNftsSuite.java

```diff
@@ -76,26 +81,293 @@ final Stream<DynamicTest> idVariantsTreatedAsExpected() {
                         .signedBy(DEFAULT_PAYER, "multiKey")));
     }
 
+    /**
+     * <a href="https://github.com/hashgraph/hedera-improvement-proposal/blob/main/HIP/hip-850.md">HIP-850</a>
+     * Tests that the transaction fails if it is not signed by the metadata key or the supply key.
+     * @return the dynamic test
+     */
     @HapiTest
-    final Stream<DynamicTest> failsIfTokenHasNoMetadataKey() {
-        return defaultHapiSpec("failsIfTokenHasNoMetadataKey")
+    final Stream<DynamicTest> failsIfNoMetadataKeyOrSupplyKey() {
+        return defaultHapiSpec("failsIfNoMetadataKeyOrSupplyKey")
                 .given(
                         newKeyNamed(SUPPLY_KEY),
-                        cryptoCreate(TOKEN_TREASURY),
+                        newKeyNamed(METADATA_KEY),
+                        cryptoCreate(TOKEN_TREASURY).maxAutomaticTokenAssociations(4),
+                        cryptoCreate(RECEIVER).maxAutomaticTokenAssociations(4),
                         tokenCreate(NON_FUNGIBLE_TOKEN)
                                 .supplyType(TokenSupplyType.FINITE)
                                 .tokenType(NON_FUNGIBLE_UNIQUE)
                                 .treasury(TOKEN_TREASURY)
+                                .maxSupply(12L)
                                 .supplyKey(SUPPLY_KEY)
+                                .metadataKey(METADATA_KEY)
```

Comment:
Probably add a test that makes token have no supply or metadata key after nfts are minted and see if it gives the new response code

Reply:
Added a test case for this.